lib/ace/mode/scrypt_highlight_rules.js

```diff
@@ -0,0 +1,719 @@
+        "#functions": [{
+            token: "meta.method.scrypt",
+            regex: /(?=[\w].*\s+)(?=(?:[^=\/]|\/(?!\/))+\()/,
+            push: [{
+                token: "meta.method.scrypt",
+                regex: /}|(?=;)/,
+                next: "pop"
+            }, {
+                include: "#contract-constructor"
+            }, {
+                include: "#contract-function"
+            }, {
+                defaultToken: "meta.method.scrypt"
+            }]
+        }],
+        "#contract-constructor": [{
+            token: [
+                "storage.type.function.scrypt",
+                "meta.function.scrypt"
+            ],
+            regex: /\b(constructor)(\s*\()/,
+            push: [{
+                token: "meta.function.scrypt",
+                regex: /\)/,
+                next: "pop"
+            }, {
+                include: "#parameters"
+            }, {
+                include: "#parens"
+            }, {
+                include: "#comments"
+            }, {
+                defaultToken: "meta.function.scrypt"
+            }]
+        }, {
+            include: "#function-body"
+        }],
+        "#contract-function": [{
+            include: "#storage-modifiers"
+        }, {
+            token: [
+                "storage.type.function.scrypt",
+                "meta.function.scrypt",
+                "entity.name.function.scrypt",
+                "meta.function.scrypt"
+            ],
+            regex: /\b(function)(\s+)([_$[:alpha:]][_$[:alnum:]]*)(\s*\()/,
+            push: [{
+                token: "meta.function.scrypt",
+                regex: /\)/,
+                next: "pop"
+            }, {
+                include: "#parameters"
+            }, {
+                include: "#parens"
+            }, {
+                include: "#comments"
+            }, {
+                defaultToken: "meta.function.scrypt"
+            }]
+        }, {
+            include: "#function-return"
+        }, {
+            include: "#function-body"
+        }, {
+            include: "#comments"
+        }],
+        "#function-body": [{
+            token: "punctuation.section.method.begin.bracket.curly.scrypt",
+            regex: /\{/,
+            push: [{
+                token: "meta.definition.function.scrypt",
+                regex: /(?=})/,
+                next: "pop"
+            }, {
+                include: "#code"
+            }, {
+                defaultToken: "meta.definition.function.scrypt"
+            }]
+        }],
+        "#function-return": [{
+            token: "keyword.operator.semicolon.scrypt",
+            regex: /\:/,
+            push: [{
+                token: "meta.method.return-type.scrypt",
+                regex: /(?=\{)/,
+                next: "pop"
+            }, {
+                include: "#comments"
+            }, {
+                token: ["text", "text", "text"],
+                regex: /\b([_$[:alpha:]][_$[:alnum:]]*)(\s*)((?:(?:\[[\w.]+\])+)?)/
```

Comment:
could you replace all usages of `_$[:alpha:]` with `\w` and `_$[:alnum:]` with `\w\d`, this seems to be a feature not supported by the translator that causes function to not be highlighted

Reply:
After fixing this, there are other problems, and the automatic generation tool may not be fully supported. I don’t use the automatic generation tool, and re-implemented the simplest version that can highlight all keywords.